utils_linux.go

```diff
@@ -121,6 +121,11 @@ func setupIO(process *libcontainer.Process, container *libcontainer.Container, c
 			if err != nil {
 				return nil, err
 			}
+			defer func() {
+				if Err != nil {
+					conn.Close()
+				}
+			}()
```

Comment:
The uc.File() call below is also missing a close.

Not sure when we can close it, though. Maybe it's handled by the caller, can you check?